### DISCUSSION TITLE
Is it possible to view the contents of the materials database of the 3D Cadwork file with Python or with a database software ?

### DISCUSSION THREAD

**Discussion by faure-levoux** [cwapi3d/cwapi3dpython#176] [General]
Hi, 
I saw that I can modify the materials database of the Cadwork Userprofil with a database software.
But the Multilayer can only use the materials in the 3D Cadwork file. 
So, I want to modify the database of the 3D Cadwork file with Python or another database software : is it possible ? 

Thanks !

**Comment by Brunner246** [comment]
Hi, 
No, it is not possible to modify the database. 
You have to create the materials via `material_controller` `crate_material...`

**Comment by Brunner246** [comment]
@Aupire 
I now understand the problem. I was not aware that only materials from the 3d file can be used for the multilayer wall.
Unfortunately, this is currently only possible with the workaround where you read the SQL database and create the materials via `material_controller`.

**Comment by faure-levoux** [comment]
Ok thanks, do you know if it will be possible to use materials from the userprofil for the multilayer wall in the future ? 
It will be a good feature.

**Comment by Brunner246** [comment]
I would also welcome the feature, but I can't tell you if this is planned in the near future.
